controllers/tab/user/ProfileTabHandler.inc.php

```diff
@@ -262,18 +263,26 @@ function changePassword($args, $request) {
 	 * @return JSONMessage JSON-formatted response
 	 */
 	function savePassword($args, $request) {
+		
 		$this->setupTemplate($request);
+
 		import('lib.pkp.classes.user.form.ChangePasswordForm');
 		$passwordForm = new ChangePasswordForm($request->getUser(), $request->getSite());
 		$passwordForm->readInputData();
 
 		if ($passwordForm->validate()) {
+
 			$passwordForm->execute();
+
+			SessionManager::logoutFromOtherDevices();
```

Comment:
This is good for the user profile tab's password edit tool, but I think you'll also have to add calls to this function everywhere else the password can be reset -- in the Journal Manager's administration interface where they can edit a password, and also in the password reset process, off the top of my head. I'd suggest looking for uses of:
- lib/pkp/classes/user/form/LoginChangePasswordForm.inc.php
- lib/pkp/controllers/grid/settings/user/form/UserDetailsForm.inc.php

Reply:
@asmecher I yes you are right and I missed few more section . But as I dig deeper into this issue based on you feedback , I think we need a way not only to invalidate other device sessions but also a method to invalidate all device sessions . Such use cases are

1. when from the journal management , an admin update another user's password , that user need to be logged out from all devices including the one that he is using right now . 

2. Another case is when user reset password from a device as that regenerate a password from the reset link comparing to the more modern and adaptive approach where user can set the password as user like . 

so I guess we need method like that can force logout other user . 

```php
SessionManager::logoutFromAllDevices(USER_ID); 
```

Reply:
I would suggest a single method to achieve both `logoutFromAllDevices` and `logoutFromOtherDevices`, with an optional session ID to exclude if it's provided. Something like e.g. `invalidateSessions(int $userId, string $excludeSessionId = null)`.

Reply:
@asmecher that's also an option and probably the better one . I thought about that but wanted to go with more of a Laravel like implementation where the method name and implementation itself is self explanatory . But having one single method to handle it better for future refactoring and enhancement . Will go with single method approach .

Reply:
I would happily merge either approach; this is stylistic. But personally I prefer to avoid multiplying out the number of methods, especially when both cases can be accommodated easily with a conditional clause (untested):
```php
DB::table('sessions')
   ->where('user_id', '=', $userId)
   ->when($excludeSessionId !== null, fn($query) => $query->where('session_id', '<>', $excludeSessionId))
   ->delete();
```
A lot of my stylistic preferences are shaped by trying to keep the codebase as small as possible over the years.